Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm

```diff
@@ -78,6 +80,30 @@ - (void)removeAllVideoTargets;
 @end
 #endif
 
+@interface WebEffectiveRateChangedListenerObjCAdapter : NSObject
+@property (nonatomic, readonly, direct) RefPtr<WebCore::EffectiveRateChangedListener> listener;
+- (instancetype)init NS_UNAVAILABLE;
+- (instancetype)initWithEffectiveRateChangedListener:(const WebCore::EffectiveRateChangedListener&)listener;
+@end
+
+NS_DIRECT_MEMBERS
+@implementation WebEffectiveRateChangedListenerObjCAdapter {
+    ThreadSafeWeakPtr<WebCore::EffectiveRateChangedListener> _listener;
+}
+
+- (instancetype)initWithEffectiveRateChangedListener:(const WebCore::EffectiveRateChangedListener&)listener
+{
+    if ((self = [super init]))
+        _listener = listener;
+    return self;
+}
+
+- (RefPtr<WebCore::EffectiveRateChangedListener>)listener
```

Comment:
If we're going to return RefPtr, our style is to call this property "protectedListener".

But I think we can just return raw pointer. Our caller stuffs the value into a RefPtr, and static analysis will check that.

Reply:
`_listener` is a `ThreadSafeWeakPtr<>`, so `get()` returns a `RefPtr<>` for thread safety.

There's no reason to change it back to a raw pointer to return it, correct?

Reply:
Sorry, I missed that detail. Since it’s thread safe recounted, the style option “protectedListener” is the way to go.

Reply:
Okay, will update the method name.  Thanks!